src/v/kafka/server/offset_translator.h

```diff
@@ -0,0 +1,74 @@
+#pragma once
+#include "kafka/types.h"
+#include "model/fundamental.h"
+#include "raft/configuration_manager.h"
+
+namespace kafka {
+/**
+ * Offset translator performs conversion between kafka offsets, which doesn't
+ * include raft configuration batches and log offsets which includes
+ */
+class offset_translator {
+public:
+    explicit offset_translator(const raft::configuration_manager& config_mgr)
+      : _cfg_mgr(config_mgr) {}
+
+    model::offset to_kafka_offset(model::offset o) const {
+        auto d = delta(o);
+        return model::offset(o - d);
+    }
+
+    model::offset from_kafka_offset(
+      model::offset o, model::offset hint = model::offset{}) const {
```

Comment:
we invoke from_kafka_offset always with only one argument, let's get rid of hint since we don't use it

Reply:
I will use it, in follow up patch